cli/spec.py

```diff
@@ -0,0 +1,260 @@
+import ConfigParser
+import os
+import yaml
+import yamlordereddictloader
+
+import clg
+from cli import exceptions
+from cli import utils
+
+SPEC_EXTENSION = '.spec'
+LOOKUP_DICT = {
+    '__DEFAULT__': "default"
+}
+
+TRIM_PARAMS = ['default', 'required', 'requires_only']
+
+
+def cfg_file_to_dict(value):
+    """
+    The custom type for clg spec
+    :param value: the argument value.
+    :return: dict based on a provided ini file.
+    """
+    config = ConfigParser.ConfigParser()
+    with open(value) as fd:
+        config.readfp(fd)
+
+    res_dict = {}
+    for section in config.sections():
+        res_dict[section] = {}
+        for option, value in config.items(section):
+            value = utils.string_to_list(value, append_to_list=False)
+            res_dict[section][option] = value
+
+        res_dict[section].pop('__name__', None)
+
+    return res_dict
+
+
+def parse_args(module_name, config, args=None):
+    """
+    Looks for all the specs for specified module
+    and parses the commandline input arguments accordingly.
+
+    :param module_name: the module name: installer|provisioner|tester
+    :param config: the infrared configuration file
+    :param args: additional arguments to pass to the clg.
+    """
+
+    res_args, subcommand_opts = _get_command_line_args(args, config,
```

Comment:
It's a bit confusing for me, try to split it into to steps:
1. get clg_opts
2. get subcommand_opt

You can try something like this:
```python
clg_options = _parse_command_line(...)
subcommand_options = _get_subcommand_options(...)
```
 Or maybe even to separate subcommand options from command options in clg_options object
```python
print clg_options
{'command_options': {...},
'subcommand_options': {...}
}
```